UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs

```diff
@@ -89,6 +192,29 @@ public void BeginExploring(Target _target)
 			target = _target;
 		}
 
+		//gets nearest available target
+		private void RefreshTargetPos()
+		{
+			if (visionCircleArea.Count < 1)
+			{
+				UpdateVisionArea();
+			}
+
+			float minDist = 999;
+			foreach(Vector3Int pos in visionCircleArea)
+			{
+				if (IsTargetFound(pos))
+				{
+					float curDist = Vector3.Distance(pos.ToWorld(mobTile.Matrix), mobTile.WorldPositionServer);
```

Comment:
You are converting this to a world coordinate but it is already a world coordinate.